lua/autorun/sh_waiter.moon

```diff
@@ -0,0 +1,82 @@
+require "moonscript"
+
+AddCSLuaFile!
+
+export Waiter
+export WaiterQueue
+
+Waiter = {}
+WaiterQueue = WaiterQueue or {}
+
+Waiter.lastLoopDuration = 0
+
+patronQueue = {}
+patronCount = 0
+
+-- Minimum delay (in seconds) between executions of attendPatrons()
+minBreakAfterAttending = 0.5
+
+generatePatron = (waitingFor, onSuccess, onTimeout, maxAttempts=0) ->
```

Comment:
Hello default args, where have you been my whole life?

Reply:
they've been on the moon